source/common/scheduler.c

```diff
@@ -165,7 +173,11 @@ DWORD scheduler_signal_waitable( HANDLE waitable, SchedularSignal signal )
 
 	dprintf( "[SCHEDULER] entering scheduler_signal_waitable( 0x%08X )", waitable );
 
+#ifdef _WIN32
 	if( schedulerThreadList == NULL || waitable == NULL )
+#else
+	if( schedulerThreadList == NULL || waitable == 0 )
+#endif

```

Comment:
This `#ifdef` could be removed by changing the code slightly. Perhaps:
``` c
if (!schedulerThreadList || !waitable)
```
Thoughts?